src/core/ext/filters/client_channel/client_channel.cc

```diff
@@ -1268,6 +1273,51 @@ static void resume_pending_batch_in_call_combiner(void* arg,
   grpc_subchannel_call_process_op(subchannel_call, batch);
 }
 
+// The callback to intercept trailing metadata if retries is not enabled
+static void recv_trailing_metadata_ready_for_lb(void* arg, grpc_error* error) {
+  subchannel_batch_data* batch_data = static_cast<subchannel_batch_data*>(arg);
+  grpc_call_element* elem = batch_data->elem;
+  call_data* calld = static_cast<call_data*>(elem->call_data);
+  GPR_ASSERT(calld->pick.recv_trailing_metadata_ready != nullptr);
```

Comment:
This assertion isn't actually necessary, since calling `GRPC_CLOSURE_SCHED()` on a null closure is a no-op.

Reply:
Changed the code to check for `calld->pick.recv_trailing_metadata != nullptr`